### PR TITLE
Issue 608 - Add option of random items in tile collection

### LINKED ISSUE #608: Add option of random items in tile collection
Like portlet collection we need an option for random items into tile collection.

### PR DESCRIPTION
**Pull request by rodfersou**
closes #608

### CODE REVIEW

File: src/collective/cover/tiles/collection.py [hunk 1]
```diff
@@ -70,6 +72,13 @@ class ICollectionTile(IPersistentCoverTile):
         default=0,
     )
 
+    form.omitted(IDefaultConfigureForm, 'random')
+    random = schema.Bool(
+        title=_(u"Select random items"),
+        required=True,
```
Comment (hvelarde) [inline src/collective/cover/tiles/collection.py]: @rodfersou why required?

File: .travis.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ env:
     - PLONE_VERSION=4.3
     - PLONE_VERSION=4.3 DEXTERITY_ONLY=true
     - PLONE_VERSION=4.3 JQUERY=1.9.1.3
-    - PLONE_VERSION=4.3 NEW_WIDGETS=true
```
Comment (hvelarde) [inline .travis.yml]: @rodfersou nice catch! I want a separated commit for this one and an update of the documentation.

**Comment by hvelarde** [comment]
@agnogueira @rodfersou I think I don't like this; isn't it a feature of the collection? why are we adding more complexity?

File: src/collective/cover/tiles/collection.py (hunk 1 above)
Comment (rodfersou) [inline src/collective/cover/tiles/collection.py]: fixed

**Comment by rodfersou** [comment]
We are following the [portlet implementation](https://github.com/plone/plone.portlet.collection/blob/ed93dbf8961dca89b1b19a54d96962804c6b0357/plone/portlet/collection/collection.py#L208) collection don't provide this feature of random items.

**Comment by davilima6** [comment]
@hvelarde, in the same manner editors may need different title and description for items it might be interesting to have a way of randomizing the items inside the tile only.

File: src/collective/cover/tests/test_collection_tile.py [hunk 3]
```diff
@@ -164,6 +164,26 @@ def test_offset(self):
...
+        self.assertEqual(
+            [r for r in self.tile.results()],
+            items_not_random
+        )
+
+        self.tile.data['random'] = True
+        random_items1 = [r for r in self.tile.results()]
+        random_items2 = [r for r in self.tile.results()]
+        random_items3 = [r for r in self.tile.results()]
+        self.assertTrue(
```
Comment (hvelarde) [inline src/collective/cover/tests/test_collection_tile.py]: @rodfersou you must avoid most logic on tests; also, there is a chance that this fails randomly.

File: src/collective/cover/tests/test_collection_tile.py [hunk 4]
```diff
@@ -164,6 +164,27 @@ def test_offset(self):
...
+        # now, return results in random order
+        self.tile.data['random'] = True
+        i = 0
+        while i < 3:
+            results = [o for o in self.tile.results()]
+            if results != ordered:
+                return
+            i += 1
+
+        self.fail('No random order after 3 attemps')
```
Comment (rodfersou) [inline src/collective/cover/tests/test_collection_tile.py]: First time I see this kind of structure in test, nice to know!

Comment (hvelarde) [inline src/collective/cover/tests/test_collection_tile.py]: that's what I was talking about; IMO, is clearer.

**Pull request merged**